compiler/vm.nim

```diff
@@ -2110,11 +2110,14 @@ proc evalStmt*(c: PCtx, n: PNode) =
   if c.code[start].opcode != opcEof:
     discard execute(c, start)
 
-proc evalExpr*(c: PCtx, n: PNode): PNode =
-  let n = transformExpr(c.graph, c.idgen, c.module, n)
-  let start = genExpr(c, n)
-  assert c.code[start].opcode != opcEof
-  result = execute(c, start)
+when false:
```

Comment:
Nitpicking but this has nothing to do with the PR.

Reply:
I've now un-disabled it but it did relate to this PR in the following weak sense: `--eval` used to (IIRC) call `evalExpr` some time ago and `--eval` became undocumented, but that link got cut at some point so `evalExpr` became deadcode; now that this PR takes over `--eval` with a new meaning, it made sense to seal the fate of this unused deadcode